pygeobase/io_base.py

```diff
@@ -312,11 +443,55 @@ def _open(self, gpi):
         cell = self.grid.gpi2cell(gpi)
         filename = os.path.join(self.path, self.fn_format.format(cell))
 
-        if self.previous_cell != cell:
-            self.close()
+        if self.mode == 'r':
+            if self._read_bulk:
```

Comment:
The rare use case is if the files are stored per grid point. Or another use case is if the ioclass does not support bulk reading, in other words as soon as the file has been opened it needs to be closed and opened again to read the data for another grid point.

But I've also no objections to make this mandatory, which basically means that each (new) ioclass has to support bulk reading.